File: AlgorithmReview/Programmers/Lv3/네트워크.swift
//
//  네트워크.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/05/05.
//

import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {

    var visit = [Bool](repeating: false, count: n)
    
    func network(_ k: Int) {
        for i in 0..<n{
            if computers[k][i] == 1 && !visit[i] {
                visit[i] = true
                network(i)
            }
        }
    }
    
    var result = 0
    for i in 0..<n{
        if !visit[i] {
            visit[i] = true
            result += 1
            network(i)
        }
    }
    
    return result
}

File: Python/네트워크.py
def solution(n, computers):
    
    visit = [False] * n
    
    def network(k):
        for i in range(n):
            if computers[k][i] == True and visit[i] == False:
                visit[i] = True
                network(i)
    
    answer = 0
    for i in range(n):
        if visit[i] == False:
            visit[i] = True
            answer += 1
            network(i)
    
    return answer
